### ISSUE TITLE
Not working on macOS Big Sur

### ISSUE THREAD

**Issue by jcarpenter** [AryToNeX/Glasstron#54]
No apparent effect on macOS Big Sur. I've tried various combinations of settings and poked through the source, but cannot get any sort of vibrancy (blurred background) effect.

**To Reproduce**
1. Run [this gist](https://gist.github.com/jcarpenter/cb82819c602eb08f0d5b14d805858a45) in Electron Fiddle, on macOS Big Sur.

**Expected behavior**
A vibrancy effect in the background of the window.

**Screenshots**

<img width="1036" alt="image" src="https://user-images.githubusercontent.com/1425497/103045006-a0e7dd80-4537-11eb-978b-4dc2d3ae2bd3.png">

**Desktop (please complete the following information):**
 - OS: macOS Big Sur 11.1
 - Electron: Tested both `9.4.0` and `11.1.1`. 
 - Glasstron: 0.0.5

**Additional context**

* This is on a 2019 MacBook Pro 15". 
* Standard vibrancy works. Albeit with a noticeable performance hit.
* I've closely reviewed the docs, but am still a little unclear on what configuration setting to use for macOS. The following is what the gist uses:

```js
const win = new glasstron.BrowserWindow({
  width: 800,
  height: 600,
  backgroundColor: 'transparent',
  // vibrancy: 'sidebar',
  // titleBarStyle: 'hiddenInset',
  transparent: true,
  webPreferences: {
    nodeIntegration: true
  }
});

// win.blurType = "sidebar";
win.setVibrancy('sidebar')
win.setBlur(true);
```

**Comment by AryToNeX** [comment]
Please clone this repo and try the included example test application.

If it works there, you might want to use the upstream code instead of the one that is currently available on NPM.

If it doesn't work, then we have a huge problem since I don't have access to Big Sur machines currently and this will likely boil down to some Apple+Electron nonsense like the rest of the stuff, really

**Comment by jcarpenter** [comment]
Thank you, I'll give that a go now. 

FWIW, after cloning the repo, the first thing I've tried is copying the BrowserWindow config object settings from `test/index.js` (below) into my Gist, and running in Fiddle.

```js
const win = new glasstron.BrowserWindow({
	width: 800,
	height: 600,
	backgroundColor: "#00000000",
	resizable: true,
	title: "Glasstron test window",
	autoHideMenuBar: true,
	frame: false, // this is a requirement for transparent windows it seems
	show: true,
	blur: true,
	blurType: "blurbehind",
	blurGnomeSigma: 100,
	blurCornerRadius: 20,
	vibrancy: "fullscreen-ui",
	webPreferences: {
		nodeIntegration: true
	}
});
```

When run, I get a vibrancy effect, but to my eyes it looks identical to the equivalent Electron vibrancy effect—complete with a rendering error in the rounded corners (documented in [this issue](https://github.com/electron/electron/issues/27080)).

![image](https://user-images.githubusercontent.com/1425497/103102219-ebb53400-45cf-11eb-9f70-fb263ff99d5f.png)

Then, when I commented out `vibrancy: "fullscreen-ui"`, it went back to a fully-transparent result:

![image](https://user-images.githubusercontent.com/1425497/103102034-15219000-45cf-11eb-918a-efc272dc3ddc.png)

I'll go ahead and try the example test application next...

**Comment by jcarpenter** [comment]
Result from running the test:

![image](https://user-images.githubusercontent.com/1425497/103102626-bc9fc200-45d1-11eb-91ce-939b4e5b1a87.png)

Console errors: 

```bash
> glasstron@0.0.5 test
> electron ./test

(node:28236) UnhandledPromiseRejectionWarning: TypeError: win.getBlur is not a function
    at IpcMainImpl.<anonymous> (/Users/josh/Desktop/GlasstronTest/Glasstron/test/index.js:62:41)
    at IpcMainImpl.emit (events.js:223:5)
    at WebContents.<anonymous> (electron/js2c/browser_init.js:173:8161)
    at WebContents.emit (events.js:223:5)
(node:28236) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:28236) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
Hitting Toggle Off has no effect, and just repeats the error
```bash
(node:28236) UnhandledPromiseRejectionWarning: TypeError: win.getBlur is not a function
    at IpcMainImpl.<anonymous> (/Users/josh/Desktop/GlasstronTest/Glasstron/test/index.js:57:42)
(node:28236) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
```

**Comment by jcarpenter** [comment]
FWIW, I've perceived a bad performance regression with vibrancy, roughly since upgrading to Big Sur (that's what motivated me to check out Glasstron). So perhaps something has indeed changed in the underlying implementation.

**Comment by AryToNeX** [comment]
Note that Glasstron will just use Electron's Vibrancy implementation on macOS, but it'll also wrap that up to be more easily controllable. If you don't plan to support anything other than macOS, then you'll be better off just using vanilla Electron plus setVibrancy. If you plan to ship your app on Windows or Linux while retaining an almost same blur effect, then Glasstron will help you achieve just that.

And FYI, corner artifacts are visible in every OS but Linux with KDE Plasma (and it was painful to support it there, also)

**Comment by jcarpenter** [comment]
> Note that Glasstron will just use Electron's Vibrancy implementation on macOS, but it'll also wrap that up to be more easily controllable. If you don't plan to support anything other than macOS, then you'll be better off just using vanilla Electron plus setVibrancy. If you plan to ship your app on Windows or Linux while retaining an almost same blur effect, then Glasstron will help you achieve just that.
> 
> And FYI, corner artifacts are visible in every OS but Linux with KDE Plasma (and it was painful to support it there, also)

Thanks for that clarification as to use-case. I am indeed Mac-only for now, so I'll forego Glasstron, for the time being.

**Comment by AryToNeX** [comment]
@jcarpenter btw all macOS issues seem to be fixed in the newest Glasstron version, 0.0.7. If you plan on shipping your next creation on more platforms than just Mac, you can do so reliably now!

**Comment by davej** [comment]
@AryToNeX Can you confirm this? Is it due to a change in Glasstron? It seems like the upstream Electron issue still isn't resolved https://github.com/electron/electron/issues/27080

**Comment by AryToNeX** [comment]
> @AryToNeX Can you confirm this? Is it due to a change in Glasstron? It seems like the upstream Electron issue still isn't resolved [electron/electron#27080](https://github.com/electron/electron/issues/27080)

If you are referring to this issue being referred into that issue, then, yes, I am afraid that window corners will be as bugged in Big Sur. But this one issue actually was about another problem, regarding this package not working at all on macOS, which is now fixed.
